protocol/app/upgrades/v4.0.0/constants.go

```diff
@@ -23,6 +24,9 @@ var Upgrade = upgrades.Upgrade{
 			// Add new ICA stores that are needed by ICA host types as of v8.
 			icacontrollertypes.StoreKey,
 			icahosttypes.StoreKey,
+
+			// Add authz module to allow granting arbitrary privileges from one account to another acocunt.
```

Comment:
Is there anything else needed in the upgrade to initialize `authz` state? Is [this](https://github.com/cosmos/cosmos-sdk/blob/eaf92c225cfa2420946ff738cdcf05cf8b402040/x/authz/genesis.go#L11) inherently left empty with no `Authorization`?

Reply:
yeah I think this should be empty. there is currently no authorizations since it was not enabled previously

Each `Authorization` consists of (granter, grantee, msgURL) and permissions needs to be granted explicitly by the granter